dev-python/argon2-cffi/argon2-cffi-21.2.0.ebuild

```diff
@@ -0,0 +1,37 @@
+# Copyright 1999-2021 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+PYTHON_COMPAT=( python3_{8..10} pypy3 )
+DISTUTILS_USE_SETUPTOOLS=pyproject.toml
+inherit distutils-r1
+
+DESCRIPTION="CFFI bindings to the Argon2 password hashing library"
+HOMEPAGE="https://github.com/hynek/argon2-cffi"
+SRC_URI="mirror://pypi/${PN:0:1}/${PN}/${P}.tar.gz"
+
+LICENSE="MIT"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+
+DEPEND=">=dev-python/argon2-cffi-bindings-21.2.0"
+RDEPEND="${DEPEND}"
+BDEPEND="
+	test? (
+		dev-python/hypothesis[${PYTHON_USEDEP}]
+	)
+"
+
+DOCS=( AUTHORS.rst CHANGELOG.md FAQ.rst README.rst )
+# Patch the tool.flit.module.name entry in pyproject.toml to include the "src" subfolder
+# This allows pyproject2setuppy to retrieve the dynamic version and description correctly
+PATCHES+=( "${FILESDIR}/${P}-pyproject-module-path.patch" )
+# Sphinx's conf.py uses importlib.metadata to determine the package version. However, argon2 is
+# not installed when we try to build the docs. Therefore, we hardcode the release version in conf.py.
+PATCHES+=( "${FILESDIR}/${P}-sphinx-importlib.patch" )
```

Comment:
This is a no-go. It's just wrong to have a patch with hardcoded version number that you'll have to redo on every bump (and somebody else will probably miss that).

Reply:
Yes, fair point.

Both patches are single line changes, so I replaced them with sed commands. This makes them a bit more robust against changes in the patched files and avoids a hardcoded version by inserting `PV` as the project version.

It will break if upstream creates a release with a PEP440 version that is incompatible with Gentoo version numbers, e.g. a `.beta0` suffix. However, the code in `conf.py` doesn't address this case and such version numbers weren't used in the past.

What do you think of this solution?

Reply:
I suppose `sed` is fine. I don't think we really need to care that much, and if things break, we can address them then.